src/seedu/addressbook/data/person/Address.java

```diff
@@ -29,6 +29,20 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.value = trimmedAddress;
     }
 
+    /**
+     * Edits current address.
+     *
+     * @param newAddress replaces current address with this address
+     * @throws IllegalValueException if given address string is invalid.
+     */
+    public void editCurrentAddress(String newAddress) throws IllegalValueException {
+        String trimmedNewAddress = newAddress.trim();
+        if (!isValidAddress(trimmedNewAddress)) {
```

Comment:
Based on the regex for addresses, it doesn't seem to have any constraints for being a valid address? There is no need to test whether the address changed is valid?